api_definition_test.go

```diff
@@ -437,3 +435,71 @@ func TestRoundRobin(t *testing.T) {
 		}
 	}
 }
+
+func setupKeepalive(conn net.Conn) error {
+	tcpConn := conn.(*net.TCPConn)
+	if err := tcpConn.SetKeepAlive(true); err != nil {
+		return err
+	}
+	if err := tcpConn.SetKeepAlivePeriod(30 * time.Second); err != nil {
+		return err
+	}
+	return nil
+}
+
+type customListener struct {
+	L net.Listener
```

Comment:
You can override methods from an embedded field - that's exactly what embedding is for. But you're right, I assumed the two interfaces were the same.